### PR TITLE
LaunchDarkly - Kosli integration

### PR DESCRIPTION
**Pull request by jumboduck**
**Requirements**

- [ ] I have followed the repository's [pull request submission guidelines](../blob/main/README.md#submitting-pull-requests)
- [ ] I have updated my user documentation
- [ ] I have updated my developer documentation (my integration README)
- [ ] I have submitted the [Become a Partner](https://launchdarkly.com/partners/) form

**Intended availability**

If you're adding a new integration, let us know if you are a community member or a partner.

If you are a partner, your integration will be available to all customers.

If you're part of the community and submitting an integration for your own purposes or internal teams, specify your company name so that we can verify your identity and enable the integration for your account.

**Related issues**

Provide links to any issues in this repository or elsewhere relating to this pull request.

**Describe the solution you've provided**

If you're adding a new integration, provide some context into how LaunchDarkly and the third-party service should interact.

If you're modifying an existing integration, provide a clear and concise description of what you expect to happen.

**Describe alternatives you've considered**

If you're modifying an existing integration, provide a clear and concise description of any alternative solutions or features you've considered.

**Provide a request preview**

Include the output from running `npm run preview <INTEGRATION_NAME>`.

**Additional context**

Add any other context about the pull request here.

### CODE REVIEW

File: integrations/kosli/manifest.json [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+        "url": "{{webhookUrl}}",
+        "method": "POST",
+        "headers": [
+          {
+            "name": "x-ld-signature",
+            "value": "{{secret}}"
+          }
+        ]
+      },
+      "templates": {
```
Comment (InTheCloudDan) [inline integrations/kosli/manifest.json:47]: Can you make `templates` and empty object `"templates": {},` and then you can remove the `templates` directory. You have `useStandardWebhookPayload` set to `true` below so no template should be needed.

File: integrations/kosli/manifest.json [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+        "default": "templates/default.json.hbs"
+      },
+      "defaultPolicy": [
+        {
+          "effect": "allow",
+          "actions": ["*"],
+          "resources": ["proj/*:env/*:flag/*"]
```
Comment (InTheCloudDan) [inline integrations/kosli/manifest.json:54]: This is going to capture all changes including metadata changes such as adding/removing tags. Do you want it that widely scoped?

**Review by InTheCloudDan — CHANGES_REQUESTED** [review]
@jumboduck just a few minor changes requested, then I can merge in and test in app.

File: integrations/kosli/manifest.json (hunk 2 above)
Comment (jumboduck) [reply integrations/kosli/manifest.json:51]: This is indeed more than we need. I've updated the actions.

File: integrations/kosli/manifest.json (hunk 1 above)
Comment (jumboduck) [reply integrations/kosli/manifest.json]: Thank you, I've updated the manifest with these changes.

**Comment by InTheCloudDan** [comment]
@jumboduck it looks good, there's one more optional piece I noticed that you may want to at least set a `defaultValue` for the endpoint: https://github.com/launchdarkly/integration-framework/blob/2b81a901bba93b5a791dc5eb98fb41337a7686bf/manifest.schema.json#L290

You can also restrict it to just a set of `allowedValues` but if you have instances or expecting to add more then you may need to add them in the future but right now each user is required to paste in the URL with no defaults.

CleanShot 2024-08-01 at 13 49 51

**Comment by jumboduck** [comment]
> @jumboduck it looks good, there's one more optional piece I noticed that you may want to at least set a `defaultValue` for the endpoint

@InTheCloudDan  We don't have a default endpoint, since the endpoint contains an identifier unique to the user as it is currently implemented on our end. Each user is provided a unique url and secret.

**Comment by meekrosoft** [comment]
@InTheCloudDan Do you require any further changes / info before merging? LMK 🤓

**Review by InTheCloudDan — APPROVED** [review]

**Comment by meekrosoft** [comment]
@InTheCloudDan can you let me know what the next steps are?  We don't see this in the integration marketplace so far...

**Pull request merged**